Calculator/Calculator/Model/CalculatorItemQueue.swift

```diff
@@ -0,0 +1,26 @@
+import Foundation
+
+struct CalculatorItemQueue<item: CalculateItem> {
+    private var enqueueList: [item] = []
+    private var dequeueList: [item] = []
+    var count: Int { dequeueList.isEmpty ? enqueueList.count : dequeueList.count }
```

Comment:
```suggestion
   var count: Int {
      return dequeueList.count + enqueueList.count
   }
```
이렇게 두 리스트 수를 다 더해줘야하지 않나요?.?

Reply:
`dequeueList`가 비어있다면 `enqueueList`에만 요소가 있을 것이고 `dequeueList`가 비어있지 않는 상황이라면 dequeue가 실행되어 `enqueueList`를 리버스 해 `dequeueList`에 할당해 주고 `enqueueList`를 비워준 상황일 것입니다.
위 근거로 `dequeueList` 가 비어있다면 `enqueueList` 의 count를, `dequeueList` 가 비어있지 않다면 `dequeueList`의 count를 할당해 주도록 구현했습니다.
아울러, 라자냐가 제안해 준 코드를 보고 생각해보았을 때 결국에는 n + 0 이거나 0 + n 이 될거라고 생각해 결과적으로는 같은 결과가 나올 것이라고 생각하는데 혹시 제가 놓친 무언가가 있을까요??
만약 결과적으로 같은 값이 나오는게 맞다면 두 코드 중에서는 어떤게 기능적으로 더 옳은 코드일지 여쭤보고 싶습니다!

Reply:
- enqueue: enqueueList로만 append
- dequeue: dequeueList로만 pop
이렇게 이해했는데요.

enqueue(1) -> enqueue(2) -> enqueue(3) -> dequeue()이렇게 호출했다면
enqueueList: empty
dequeueList: 2, 1 (3은 pop되었음)
이렇게 들어가 있을거 같은데요.

이 상태에서 enqueue(4)를 호출한다면
enqueueList: 4
dequeueList: 2, 1
이렇게 있지 않을까요?

Reply:
이번 프로젝트인 계산기 어플의 경우 `=` 을 누르면 dequeue가 바로 실행되서 요소들이 다 빠져나가기 전에(연산이 완료되기 전에)는 enqueue를 할 일이 없을 것이라고 생각해 위와 같이 작성했는데 계산기라는 생각을 제외하고 queue라는 자체만 놓고 본다면 넣었다 뺐다 할 수 있는 거니 제안해 주신 코드가 더 적합해 보입니다
아울러, " `=` 을 누르면 전부 dequeue가 될 것이다!" 라는 생각은 다음 스텝이 진행돼 봐야 더 고민해 볼 수 있는 내용이므로 현재 상황에서는 라자냐가 제안해 준 코드로 변경할 예정입니다!